### ISSUE TITLE
ubuntu repository error messages

### ISSUE THREAD

**Issue by johngheim** [Macaulay2/M2#499]
I am getting errors when trying to use the ubuntu xenial repository for macaulay2:
# apt-get update
[...]
W: http://www.math.uiuc.edu/Macaulay2/Repositories/Ubuntu/dists/xenial/Release.gpg: Signature by key CD9C0E09B0C780943A1AD85553F8BD99F40DCB31 uses weak digest algorithm (SHA1)
W: Invalid 'Date' entry in Release file /var/lib/apt/lists/www.math.uiuc.edu_Macaulay2_Repositories_Ubuntu_dists_xenial_Release

**Comment by DanGrayson** [comment]
I think "W" indicates it's just a warning message.  Did it install, anyway?

**Comment by johngheim** [comment]
The install works but you have to pass the --allow-unauthenticated flag. 
In other words, it's the same as doing the install without the gpg keys 
at all.

In maintaining my own repositories, I've found that to fix this problem, 
you need to add the following lines to your ~/.gnupg/gpg.conf file  and 
then regenerate the key:

personal-digest-preferences SHA256
cert-digest-algo SHA256
default-preference-list SHA512 SHA384 SHA256 SHA224 AES256 AES192 AES 
CAST5 ZLIB BZIP2 ZIP Uncompressed









On 07/19/2016 09:40 PM, Daniel R. Grayson wrote:
> I think "W" indicates it's just a warning message.  Did it install, anyway?
>
> ---
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/Macaulay2/M2/issues/499#issuecomment-233823503

-- 
John Heim
john@johnheim.com

**Comment by DanGrayson** [comment]
I just tried apt-get update under xenial, got the same warning messages, but it worked fine without passing --allow-unauthenticated to it.  So I can't reproduce your problem.

But I've fixed it, as suggested, see https://github.com/Macaulay2/Macaulay2-web-site/commit/aa417bedefd2ceeff831048ad4a62bda11d46250

**Comment by DanGrayson** [comment]
Let me know whether it works.

**Comment by johngheim** [comment]
How do I get the new public key? The keys generated in 2010 are still on 
http://www.math.uiuc.edu/Macaulay2/PublicKeys/.




On 07/23/2016 07:58 AM, Daniel R. Grayson wrote:
> Let me know whether it works.
>
> ---
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/Macaulay2/M2/issues/499#issuecomment-234716954

-- 
John Heim
john@johnheim.com

**Comment by DanGrayson** [comment]
There is no new public key.  We're still using the same one.

**Comment by DanGrayson** [comment]
There is no new private key, either.  No keys have been created lately.

Oh, I forgot to mention something -- you probably have to remove every trace of
release files from math.uiuc.edu in /var manually, and maybe also the Release.gpg files.
I seem to recall using something like 'find /var -name \*math.uiuc.edu*' to find them.

Let's get this to work for you properly.

**Comment by johngheim** [comment]
Yeah, that got rid of the messages. So apt musthave a protocol for 
determining whether it has to download new release files. If you use 
that find command to remove the uiuc.edu files, and then run 'apt-get 
update', it says it's getting them. If you run it again, it just said it 
hit your repository. So the second time, it must say "I've already got 
those files."

So now even the apt warnings are gone. I put macaulay2 back into the 
automatic install system yesterday and it worked. So my problem is 
solved. But I would imagine that while most other macaulay2 users 
wouldn't have had a problem as severe as mine, they'd still have gotten 
those warnings about they key being insecure. That would have been more 
of an annoyance than a problem. I'm still getting the warning for a lot 
of other repositories like Dell's and theUW's condor package repository. 
I maintain a private repository for packages that have .deb files but no 
repository (like skype) so they can still be part of the auto-install 
system. But when I got that warning on my own repository, I just 
generated whole new keys. I guess I have a lot to learn about gpg.

Thanks.

On 07/26/2016 02:15 PM, Daniel R. Grayson wrote:
> There is no new private key, either.  No keys have been created lately.
>
> Oh, I forgot to mention something -- you probably have to remove every trace of
> release files from math.uiuc.edu in /var manually, and maybe also the Release.gpg files.
> I seem to recall using something like 'find /var -name \*math.uiuc.edu*' to find them.
>
> Let's get this to work for you properly.
>
> ---
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/Macaulay2/M2/issues/499#issuecomment-235374418

-- 
John Heim
john@johnheim.com
